From 46d185b7a9b47c900760e0fb780cf8752c7e7148 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Thu, 13 May 2021 16:22:18 +0200
Subject: [PATCH 1/1] sandbox: add symbol _init for RISC-V compilation

The sandbox does not build on RISC-V when _init is not defined. Errors like
the following were observed. Which function was referred to depended on the
configuration.

    /usr/bin/ld: common/built-in.o: in function `parse_stream_outer':
    /common/cli_hush.c:3175: undefined reference to `_init'
    collect2: error: ld returned 1 exit status
    make: *** [Makefile:1726: u-boot] Error 1

    /usr/bin/ld: common/built-in.o: in function `bootdelay_process':
    common/autoboot.c:335: undefined reference to `_init'
    collect2: error: ld returned 1 exit status
    make: *** [Makefile:1726: u-boot] Error 1

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 arch/sandbox/cpu/u-boot.lds | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/sandbox/cpu/u-boot.lds b/arch/sandbox/cpu/u-boot.lds
index a1f509c9ab..d9e7ffcea3 100644
--- a/arch/sandbox/cpu/u-boot.lds
+++ b/arch/sandbox/cpu/u-boot.lds
@@ -9,6 +9,8 @@ SECTIONS
 {
 
 	. = ALIGN(4);
+	/* RISC-V GCC wants this dummy symbol */
+	_init = .;
 	.u_boot_list : {
 		KEEP(*(SORT(.u_boot_list*)));
 	}
-- 
2.30.2

